grpc-server/hardware.go

```diff
@@ -75,30 +69,19 @@ func (s *server) Push(ctx context.Context, in *hardware.PushRequest) (*hardware.
 	}
 
 	s.watchLock.RLock()
-	if ch := s.watch[h.ID]; ch != nil {
+	if ch := s.watch[hw.Id]; ch != nil {
 		select {
-		case ch <- in.Data:
+		case ch <- string(data):
 		default:
 			metrics.WatchMissTotal.Inc()
-			logger.With("id", h.ID).Info("skipping blocked watcher")
+			logger.With("id", hw.Id ).Info("skipping blocked watcher")
 		}
 	}
 	s.watchLock.RUnlock()
```

Comment:
Logging is slow, it should not be done between watch lock/unlock. Critical logging for error case is fine. But for information logging, we should do it after unlock.